In [1]:
# This Python 3 environment comes with many helpful analytics libraries installed
# It is defined by the kaggle/python Docker image: https://github.com/kaggle/docker-python
# For example, here's several helpful packages to load

import numpy as np # linear algebra
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)

# Input data files are available in the read-only "../input/" directory
# For example, running this (by clicking run or pressing Shift+Enter) will list all files under the input directory

import os
for dirname, _, filenames in os.walk('/kaggle/input'):
    for filename in filenames:
        print(os.path.join(dirname, filename))

# You can write up to 20GB to the current directory (/kaggle/working/) that gets preserved as output when you create a version using "Save & Run All" 
# You can also write temporary files to /kaggle/temp/, but they won't be saved outside of the current session

/kaggle/input/titanic/train.csv
/kaggle/input/titanic/test.csv
/kaggle/input/titanic/gender_submission.csv


In [2]:
df = pd.read_csv("/kaggle/input/titanic/train.csv")
df.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

# droping some features those are less important as per basic intuition.
**making model simple but effective**

In [3]:
df.drop(['PassengerId','Name','SibSp','Parch','Ticket','Cabin','Embarked'],axis='columns',inplace=True)
df.head()

Survived  Pclass     Sex   Age     Fare
0         0       3    male  22.0   7.2500
1         1       1  female  38.0  71.2833
2         1       3  female  26.0   7.9250
3         1       1  female  35.0  53.1000
4         0       3    male  35.0   8.0500

**seeting dependent and independent variable**

In [4]:
inputs = df.drop('Survived',axis='columns')
target = df.Survived

In [5]:
inputs.head()

Pclass     Sex   Age     Fare
0       3    male  22.0   7.2500
1       1  female  38.0  71.2833
2       3  female  26.0   7.9250
3       1  female  35.0  53.1000
4       3    male  35.0   8.0500

**Here we have to sex column into numerical.
This can be done many ways:**

In [6]:
# inputs.Sex = inputs['Sex'].replace({'male': 1, 'female': 0})
# inputs.Sex

**Get dummies method pandas**

In [7]:
dummies = pd.get_dummies(inputs.Sex  )
dummies.head()

female  male
0       0     1
1       1     0
2       1     0
3       1     0
4       0     1

In [8]:
inputs = pd.concat([inputs,dummies],axis='columns')
inputs.head(3)

Pclass     Sex   Age     Fare  female  male
0       3    male  22.0   7.2500       0     1
1       1  female  38.0  71.2833       1     0
2       3  female  26.0   7.9250       1     0

**I am dropping male column as well because of dummy variable trap theory. One column is enough to repressent male vs female**

In [9]:
inputs.drop(['Sex','male'],axis='columns',inplace=True)
inputs.head(3)

Pclass   Age     Fare  female
0       3  22.0   7.2500       0
1       1  38.0  71.2833       1
2       3  26.0   7.9250       1

# checking missing values

In [10]:
inputs.isnull().sum()

Pclass      0
Age       177
Fare        0
female      0
dtype: int64

******here age has 177 missing values. Filling missing values with meadian value of age column.**

In [11]:
inputs['Age'].median()

28.0

In [12]:
inputs.Age = inputs.Age.fillna(inputs.Age.median())
inputs.head()

Pclass   Age     Fare  female
0       3  22.0   7.2500       0
1       1  38.0  71.2833       1
2       3  26.0   7.9250       1
3       1  35.0  53.1000       1
4       3  35.0   8.0500       0

# importing and building model

In [13]:
from sklearn.naive_bayes import GaussianNB
model = GaussianNB()

In [14]:
model.fit(inputs, target)

GaussianNB()

In [15]:
test_data1 = pd.read_csv("/kaggle/input/titanic/test.csv")

In [16]:
test_data1.head()

PassengerId  Pclass                                          Name     Sex  \
0          892       3                              Kelly, Mr. James    male   
1          893       3              Wilkes, Mrs. James (Ellen Needs)  female   
2          894       2                     Myles, Mr. Thomas Francis    male   
3          895       3                              Wirz, Mr. Albert    male   
4          896       3  Hirvonen, Mrs. Alexander (Helga E Lindqvist)  female   

    Age  SibSp  Parch   Ticket     Fare Cabin Embarked  
0  34.5      0      0   330911   7.8292   NaN        Q  
1  47.0      1      0   363272   7.0000   NaN        S  
2  62.0      0      0   240276   9.6875   NaN        Q  
3  27.0      0      0   315154   8.6625   NaN        S  
4  22.0      1      1  3101298  12.2875   NaN        S

In [17]:
test_data1.shape

(418, 11)

**making X_test and y_test**

In [18]:
test_data = test_data1.drop(['PassengerId','Name','SibSp','Parch','Ticket','Cabin','Embarked'],axis='columns')
test_data.head()

Pclass     Sex   Age     Fare
0       3    male  34.5   7.8292
1       3  female  47.0   7.0000
2       2    male  62.0   9.6875
3       3    male  27.0   8.6625
4       3  female  22.0  12.2875

In [19]:
dummies_test = pd.get_dummies(test_data.Sex  )
dummies_test.head()

female  male
0       0     1
1       1     0
2       0     1
3       0     1
4       1     0

In [20]:
X_test = pd.concat([test_data, dummies_test],axis='columns')
X_test.drop(['Sex','male'],axis='columns',inplace=True)
X_test.head()

Pclass   Age     Fare  female
0       3  34.5   7.8292       0
1       3  47.0   7.0000       1
2       2  62.0   9.6875       0
3       3  27.0   8.6625       0
4       3  22.0  12.2875       1

In [21]:
X_test.isnull().sum()

Pclass     0
Age       86
Fare       1
female     0
dtype: int64

In [22]:
X_test.Fare.mode()

0    7.75
dtype: float64

In [23]:
X_test.Age = X_test.Age.fillna(X_test.Age.median())
X_test.Fare = X_test.Fare.fillna(X_test.Fare.mean())
X_test.head()

Pclass   Age     Fare  female
0       3  34.5   7.8292       0
1       3  47.0   7.0000       1
2       2  62.0   9.6875       0
3       3  27.0   8.6625       0
4       3  22.0  12.2875       1

In [24]:
X_test[X_test['Fare'].isnull()]

Empty DataFrame
Columns: [Pclass, Age, Fare, female]
Index: []

In [25]:
y_predicted = model.predict(X_test)
y_predicted

array([0, 1, 0, 0, 1, 0, 1, 0, 1, 0, 0, 0, 1, 0, 1, 1, 0, 0, 1, 1, 0, 0,
       1, 0, 1, 0, 1, 0, 0, 0, 0, 0, 1, 1, 0, 0, 1, 1, 0, 0, 0, 0, 0, 1,
       1, 0, 0, 0, 1, 1, 0, 0, 1, 1, 0, 0, 0, 0, 0, 1, 0, 0, 0, 1, 1, 1,
       1, 0, 0, 1, 1, 0, 1, 0, 1, 1, 0, 1, 0, 1, 0, 1, 0, 0, 0, 0, 1, 1,
       1, 0, 1, 0, 1, 0, 0, 0, 1, 0, 1, 0, 1, 0, 0, 0, 1, 0, 0, 0, 0, 0,
       0, 1, 1, 1, 1, 0, 0, 1, 1, 1, 1, 0, 1, 0, 0, 1, 0, 1, 0, 0, 0, 0,
       1, 0, 0, 0, 0, 0, 1, 0, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 1,
       0, 0, 1, 1, 0, 1, 1, 0, 1, 0, 0, 1, 0, 0, 1, 1, 0, 0, 0, 0, 0, 1,
       1, 0, 1, 1, 0, 1, 1, 0, 1, 0, 1, 0, 1, 0, 0, 0, 0, 0, 0, 0, 1, 1,
       0, 1, 1, 0, 1, 1, 0, 0, 1, 0, 1, 0, 0, 0, 0, 1, 1, 0, 1, 1, 1, 0,
       1, 0, 1, 0, 1, 1, 0, 1, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 1, 1, 1, 1,
       1, 0, 0, 0, 1, 0, 1, 1, 1, 0, 1, 0, 0, 0, 0, 0, 1, 0, 0, 0, 1, 1,
       0, 0, 0, 0, 1, 0, 1, 0, 1, 1, 0, 1, 0, 0, 0, 0, 1, 0, 1, 1, 1, 0,
       0, 1, 0, 0, 0, 1, 0, 1, 0, 0, 1, 0, 0, 0, 0,

In [26]:
test_data1['PassengerId']

0       892
1       893
2       894
3       895
4       896
       ... 
413    1305
414    1306
415    1307
416    1308
417    1309
Name: PassengerId, Length: 418, dtype: int64

In [27]:
submission_dataframe = pd.DataFrame(data = test_data1['PassengerId'], columns = ['PassengerId'])

In [28]:
submission_dataframe['Survived'] = y_predicted
submission_dataframe

PassengerId  Survived
0            892         0
1            893         1
2            894         0
3            895         0
4            896         1
..           ...       ...
413         1305         0
414         1306         1
415         1307         0
416         1308         0
417         1309         0

[418 rows x 2 columns]

In [29]:
submission_dataframe.to_csv('submission.csv', index = False)